### PR TITLE
Update documentation URLs for security and privacy

### PR DESCRIPTION
**Pull request by davidjb**
## Proposed change

This updates core documentation URLs to their secure counterparts for improving security and privacy.  Everything still functions the same, but the cloning of development HA now occurs over https (`git://` is insecure), mention of a specific YAML linter is now one served over HTTPS and one that operates in browser (rather than POSTing), and several locations where resources were accessed insecurely (e.g. test mosquitto certificate) have been updated also.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

Continuation of efforts in #19607

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/tools/dev-tools.markdown [hunk 1]
```diff
@@ -110,4 +110,4 @@ The Template Editor provides a way to quickly test templates prior to placing th
 
 By default this will contain sample code that illustrates how templates can be written and tested. This sample code can be removed and replaced with your own. You can restore the default example by pressing the "Reset to Demo Template" button beneath the code editor.
 
-For more information about Jinja2, visit [Jinja2 documentation](http://jinja.pocoo.org/docs/dev/templates/), and also read templating document [here](/topics/templating/).
+For more information about Jinja2, visit [Jinja2 documentation](ihttps://jinja.palletsprojects.com/templates/), and also read templating document [here](/topics/templating/).
```
Comment (frenck) [inline source/_docs/tools/dev-tools.markdown]:
```suggestion
For more information about Jinja2, visit [Jinja2 documentation](https://jinja.palletsprojects.com/templates/), and also read templating document [here](/topics/templating/).
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/tools/dev-tools.markdown (hunk 1 above)
Comment (davidjb) [reply source/_docs/tools/dev-tools.markdown]: Doh, nice catch.  Me and my Vim fingers.  Change made and pushed 👍

**Comment by frenck** [comment]
@davidjb For the future, please don't squash and force push. I have to review the full PR again now (instead of just the change).

**Comment by davidjb** [comment]
@frenck Many apologies for that - thanks for letting me know.  I'll make sure commits are separate next time.

**Review by frenck — APPROVED** [review]
Thanks, @davidjb 👍 

Happy Hacktoberfest 🎉 

../Frenck

**Pull request merged**